### PR TITLE
Add borders, advanced shadows, and hovers

### PR DESCRIPTION
**Pull request by adebroux**
## Problem
We want border and advanced shadow controls.

## Solution
Add border and advanced shadow controls for a new custom button type.

### CODE REVIEW

**Comment by adebroux** [comment]
[[Buttons] Allow makers to set more styles than they can today](https://trello.com/c/wAXcvq1G/1600-buttons-allow-makers-to-set-more-styles-than-they-can-today)

File: src/TextButton/TextButton.js [hunk 1]
```diff
@@ -64,8 +66,24 @@ export default class WrappedTextButton extends Component {
...
       const opacityValue = typeof opacity === 'number' ? opacity : 100
+      const borderStyles =
+        border?.borderStyle && border?.borderStyle !== 'none' ? border : {}
+
+      let shadowStyles = {}
+
+      if (advancedShadow?.enabled) {
+        shadowStyles = {
+          shadowColor: advancedShadow.color,
+          shadowOffset: { width: advancedShadow.x || 0, height: advancedShadow.y || 0 },
```
Comment (enzomonjardin) [inline src/TextButton/TextButton.js:77]:
```suggestion
          shadowOffset: { width: advancedShadow.x ?? 0, height: advancedShadow.y ?? 0 },
```

File: src/TextButton/TextButton.js [hunk 2]
```diff
@@ -64,8 +66,24 @@ export default class WrappedTextButton extends Component {
...
+          shadowOpacity: 1,
+          shadowRadius: advancedShadow.size || 0,
```
Comment (enzomonjardin) [inline src/TextButton/TextButton.js:79]:
```suggestion
          shadowRadius: advancedShadow.size ?? 0,
```

**Review by enzomonjardin — APPROVED** [review]
LGTM! Left a suggestion.

**Comment by adebroux** [comment]
[[Buttons] Add hover effect for web buttons only](https://trello.com/c/XkPX9bxW/1601-buttons-add-hover-effect-for-web-buttons-only)

**Pull request merged**